sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -76,14 +76,34 @@ static void _sleep_radio(gnrc_lorawan_t *mac)
     dev->driver->set(dev, NETOPT_STATE, &state, sizeof(state));
 }
 
-void gnrc_lorawan_init(gnrc_lorawan_t *mac, uint8_t *nwkskey, uint8_t *appskey)
+#if IS_ACTIVE(CONFIG_LORAWAN_1_1)
+void gnrc_lorawan_init_11x(gnrc_lorawan_t *mac, uint8_t *joineui,
```

Comment:
I think at this point it makes sense to use a common `gnrc_lorawan_init` function but with a "ctx" parameter. E.g
```c
void gnrc_lorawan_init(gnrc_lorawan_t *mac, const gnrc_lorawan_ctx_t *ctx)
{
   mac->nwkskey = ctx->nwskey;
    ....
}
```
Since this function would be resolved at compile time (e.g user cannot switch between 1.0.3 and 1.1 on runtime), I think it will simplify things a bit

Reply:
Is there a kind of hard cut between `gnrc_netif_lorawan` and `gnrc_lorawan` ? Otherwise I could just pass in a pointer to `gnrc_netif_lorawan_t` without needed to declare a new ctx struct ?

Reply:
Okay I see :) I added the `gnrc_lorawan_ctx_t` struct.

Another question regarding `gnrc netif`: Right now to configure the NwkSKey for Lorawan 1.0x one has to set `CONFIG_LORAMAC_FNWKSINT_KEY_DEFAULT`. As I think this is kind of confusing for someone wanting to configure a 1.0x node I think it would be better to use the old 1.0x terms in the netif layer and only switch to the new terms internally ?

Reply:
hmmm good question... I usually tend to think that's better to reflect the actual key names depending on the version. For example, a user that explicitly wants to use LoRaWAN 1.1 will probably know the key names. Most LoRaWAN users will still use LoRaWAN 1.0.x, so it's probably to keep the original names there. You can always fallback to macro aliases that use the terms internally.